### PR TITLE
Update pre commit hooks to run on changed files

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

Currently our precommit hooks grep a list of staged files and run various checks (black/flask8/codespell). When the string containing staged files is empty it runs these checks on all files in the project/subdirectory. This change will include a check to validate whether the $staged_files string is empty.  

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: hooks/scripts/run-codespell.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/bin/sh
-staged_files=$(git diff --staged --name-only HEAD --diff-filter=ACMR | grep -E '\.py$')
```
Comment (mabdinur) [inline hooks/scripts/run-codespell.sh:2]: Run codespell on all staged files, not just py files

**Review by Yun-Kim — COMMENTED** [review]
It would probably be useful to add some message that explains that the check was skipped because no python files were detected in staging

File: hooks/scripts/run-codespell.sh [hunk 2]
```diff
@@ -1,3 +1,7 @@
 #!/bin/sh
 staged_files=$(git diff --staged --name-only HEAD --diff-filter=ACMR)
-riot -v run -s hook-codespell $staged_files
+if [ -z "$staged_files" ]; then
+    riot -v run -s hook-codespell $staged_files
+else
+    echo 'Run codespell skipped: No Python files were found in git diff --staged'
```
Comment (Yun-Kim) [inline hooks/scripts/run-codespell.sh:6]:
```suggestion
```
Don't think this should run if there's no files in staging

**Review by Yun-Kim — CHANGES_REQUESTED** [review]
LGTM other than the one comment!

File: hooks/scripts/run-codespell.sh (hunk 2 above)
Comment (mabdinur) [reply hooks/scripts/run-codespell.sh:6]: I made one more change. Codespell is now run on all files so I changed `echo 'Run codespell skipped: No Python files were found in git diff --staged'` to `echo 'Run codespell skipped: No files were found in git diff --staged'`

**Review by Yun-Kim — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Comment by P403n1x87** [comment]
Shouldn't the `-z`s be `-n`s? 🤔

**Pull request merged**